spec/controllers/songs_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+# that an instance is receiving a specific message.
+
+RSpec.describe SongsController, :type => :controller do
+
+  # This should return the minimal set of attributes required to create a valid
+  # Song. As you add validations to Song, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+    skip("Add a hash of attributes valid for your model")
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # SongsController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe "GET index" do
+    it "assigns all songs as @songs" do
+      song = Song.create! valid_attributes
+      get :index, {}, valid_session
+      expect(assigns(:songs)).to eq([song])
+    end
+  end
+
+  describe "GET show" do
+    it "assigns the requested song as @song" do
+      song = Song.create! valid_attributes
+      get :show, {:id => song.to_param}, valid_session
+      expect(assigns(:song)).to eq(song)
+    end
+  end
+
+  describe "GET new" do
+    it "assigns a new song as @song" do
+      get :new, {}, valid_session
+      expect(assigns(:song)).to be_a_new(Song)
+    end
+  end
+
+  describe "GET edit" do
+    it "assigns the requested song as @song" do
+      song = Song.create! valid_attributes
+      get :edit, {:id => song.to_param}, valid_session
+      expect(assigns(:song)).to eq(song)
+    end
+  end
+
+  describe "POST create" do
+    describe "with valid params" do
+      it "creates a new Song" do
+        expect {
+          post :create, {:song => valid_attributes}, valid_session
+        }.to change(Song, :count).by(1)
+      end
+
+      it "assigns a newly created song as @song" do
+        post :create, {:song => valid_attributes}, valid_session
+        expect(assigns(:song)).to be_a(Song)
+        expect(assigns(:song)).to be_persisted
+      end
+
+      it "redirects to the created song" do
+        post :create, {:song => valid_attributes}, valid_session
+        expect(response).to redirect_to(Song.last)
+      end
+    end
+
+    describe "with invalid params" do
+      it "assigns a newly created but unsaved song as @song" do
+        post :create, {:song => invalid_attributes}, valid_session
+        expect(assigns(:song)).to be_a_new(Song)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, {:song => invalid_attributes}, valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT update" do
+    describe "with valid params" do
+      let(:new_attributes) {
+        skip("Add a hash of attributes valid for your model")
+      }
+
+      it "updates the requested song" do
+        song = Song.create! valid_attributes
+        put :update, {:id => song.to_param, :song => new_attributes}, valid_session
```

Comment:
Line is too long. [83/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.